locust_plugins/jmeter_listener.py

```diff
@@ -0,0 +1,242 @@
+                status_code = str(response.status_code)
+                thread_name = self.user_name
+                elapsed_time = str(round(response.elapsed.microseconds / 1000))
+                response_time = elapsed_time
+                response_length = str(len(response.text))
+                if response.ok:
+                    response_message = "OK"
+                    success = "true"
+                    exception = ""
+                else:
+                    response_message = "KO"
+                    success = "false"
+                    exception = str(response.reason)
+
+                binary_codecs = [
+                    "base64",
+                    "base_64",
+                    "bz2",
+                    "hex",
+                    "quopri",
+                    "quotedprintable",
+                    "quoted_printable",
+                    "uu",
+                    "zip",
+                    "zlib",
+                ]
+                data_type = "binary" if response.encoding in binary_codecs else "text"
+                bytes_sent = "0"
+                group_threads = str(self.user_count)
+                all_threads = str(self.runner.user_count)
+                latency = "0"
+                idle_time = "0"
+                connect = "0"
+
+                self.add_result(
+                    timestamp,
+                    response_time,
+                    name,
+                    status_code,
+                    response_message,
+                    thread_name,
+                    data_type,
+                    success,
+                    exception,
+                    response_length,
+                    bytes_sent,
+                    group_threads,
+                    all_threads,
+                    latency,
+                    idle_time,
+                    connect,
+                )
+            except:
+                logging.error("failed to log result")
+            return response
+
+        return wrapper
+
+    def add_result(
+            self,
+            timestamp,
+            response_time,
+            name,
+            status_code,
+            response_message,
+            thread_name,
+            data_type,
+            success,
+            exception,
+            response_length,
+            bytes_sent,
+            group_threads,
+            all_threads,
+            latency,
+            idle_time,
+            connect,
+    ):
+        row = [
+            timestamp,
+            response_time,
+            name,
+            status_code,
+            response_message,
+            thread_name,
+            data_type,
+            success,
+            exception,
+            response_length,
+            bytes_sent,
+            group_threads,
+            all_threads,
+            latency,
+            idle_time,
+            connect,
+        ]
+        if len(self.csv_results) >= self.flush_size:
+            self._flush_to_log()
```

Comment:
you should probably flush *after* adding the new record to the log instead of before

Reply:
Hmm... Unfortunately Black kind of massacred your function (because of the long parameter list). Maybe consider inlining it in _add_record instead.

Reply:
I have re-jigged this to accommodate what is received by the event hooks and allowing for possible additional http relevant parameters